host-configs/lassen-blueos_3_ppc64le_ib_p9-clang@upstream_gfortran-cuda.cmake

```diff
@@ -0,0 +1,82 @@
+####################################################################
+# Generated host-config - Edit at own risk!
+####################################################################
+# Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+# other Serac Project Developers. See the top-level LICENSE file for
+# details.
+#
+# SPDX-License-Identifier: (BSD-3-Clause) 
+####################################################################
+
+#---------------------------------------
+# SYS_TYPE: blueos_3_ppc64le_ib_p9
+# Compiler Spec: clang@upstream_gfortran
+# CMake executable path: /usr/tce/packages/cmake/cmake-3.18.0/bin/cmake
+#---------------------------------------
+
+#---------------------------------------
+# Compilers
+#---------------------------------------
+set(CMAKE_C_COMPILER "/usr/tce/packages/clang/clang-upstream-2019.08.15/bin/clang" CACHE PATH "")
+
+set(CMAKE_CXX_COMPILER "/usr/tce/packages/clang/clang-upstream-2019.08.15/bin/clang++" CACHE PATH "")
+
+set(CMAKE_C_FLAGS "--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1" CACHE PATH "")
+
+set(CMAKE_CXX_FLAGS "--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1" CACHE PATH "")
+
+set(BLT_EXE_LINKER_FLAGS " -Wl,-rpath,/usr/tce/packages/gcc/gcc-8.3.1/lib" CACHE PATH "Adds a missing libstdc++ rpath")
+
+#------------------------------------------------------------------------------
+# Cuda
+#------------------------------------------------------------------------------
+
+set(ENABLE_CUDA ON CACHE BOOL "")
+
+set(CUDA_TOOLKIT_ROOT_DIR "/usr/tce/packages/cuda/cuda-11.0.182" CACHE PATH "")
+
+set(CMAKE_CUDA_COMPILER "${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc" CACHE PATH "")
```

Comment:
I would expect these to be populated by `find_package(CUDA REQUIRED)`, but I guess it doesn't hurt to be explicit

Reply:
BLT's CUDA supports started before what is currently is CMake and also supports separable compiliation which the current CMake functionality does not support yet.  Because of this, we have a mix between FindCUDA (deprecated but still necessary) and the CMake first class citizen language support.

It is possible its not needed though...

Reply:
CUDA_SEPARABLE_COMPILATION has been in CMake since 3.6, it seems

Reply:
whether or not it works is another question!

Reply:
It was in FindCUDA CMake since then, but it is still not supported in the new CUDA CMake support, see: https://cmake.org/cmake/help/v3.18/release/3.18.html#languages

Reply:
I think that only applies to compiling CUDA with clang directly (CUDA language support in clang lags behind nvcc considerably). Separable compilation works fine with CMake + nvcc in my experience.

Reply:
find_package(CUDA) is NOT the new cmake CUDA as a first case language capability.

There are several codes at the lab that use BLT's CUDA support that relies on a mixture of both that utilize the CUDA separable compilation but that specifically relies on the deprecated find_package(CUDA)